hydrolib/core/io/fnm/serializer.py

```diff
@@ -2,21 +2,20 @@
 
 import inspect
 from pathlib import Path
-from typing import Dict, Iterable, Optional
+from typing import Dict, Iterable, Optional, Union
 
 
-def _calculate_max_value_length(data: Iterable[Optional[Path]]) -> int:
-    sizes = (len(str(v)) for v in data if v is not None)
-    return max(sizes)
+def _calculate_max_value_length(data: Iterable[str]) -> int:
+    return max(map(len, data))
 
 
-def _calculate_offset(value: str, max_len: int) -> str:
-    return (max_len - len(value)) * " "
+def _get_string_value(path_value: Optional[Union[dict, Path]]) -> str:
+    if type(path_value) is dict:
+        value = str(path_value.get("filepath", ""))
+    else:
+        value = str(path_value) if path_value is not None else ""
```

Comment:
Personally I think you should be more direct, you are expecting an object type Path, so:
```suggestion
    value: str = "" # Initialize to empty string.
    if type(path_value) is dict:
        value = str(path_value.get("filepath", ""))
    elif isinstance(path, Path):
        value = path_value.resolve()        
```

Of course it would be good to have tests covering this part.

Reply:
Also as a heads up, I recall the approach of hydrolib was a 'hard fail'. With the above code we do not allow it to crash.

Reply:
Thanks @Carsopre , I've implemented this, but also added an ``elif isinstance(path_value, str):`, because that was needed for references to non-existent files in some unit tests. See 7eaebb2.

Covered by test `test_parse_serialize_should_return_same_result()`